### DISCUSSION TITLE
PGP help document

### DISCUSSION THREAD

**Discussion by asterysk** [simple-login/app#468] [Feature Requests]
May I suggest that a more comprehensive write up on using PGP is added to the faq section and also the blog updated.

It is very complicated and it would be useful to see all the possible Use Cases, from my testing it seems pretty unworkable at the moment for anything other than the SimpleLogin forwarding to Protonmail PGP hop.  Its possible, however, that I haven't found the workable ways to expand beyond this hence why I'm suggesting a comprehensive write up

**Comment by nguyenkims** [comment]
Hey what do you think we can write about? The PGP is supported in both forwarding and sending phase:
- By adding a PGP key to your mailbox, an email is encrypted before being forwarded to your mailbox
- By adding a PGP key to a contact (aka reverse-alias), when you send an email to this reverse-alias, the email will be encrypted before being sent from your alias.

**Reply by leshik** [reply]
I would love to have a third option – adding a PGP key to an alias, an email is encrypted before being forwarded to the mailbox, but only for this alias. Where can I suggest such a feature?

**Reply by nguyenkims** [reply]
You can add a PGP key for an alias and send that key to your contact (or publish it) without SimpleLogin. This is how we did for our support address which also supports PGP https://simplelogin.io/pgp-key/

**Reply by leshik** [reply]
What I want is to get unencrypted mail to an alias (which terminates at SimpleLogin), and then it being forwarded to the mailbox PGP encrypted (to protect mail sent to this alias from the eyes of Gmail). How is that can be done without using SimpleLogin when MX records point to it?

**Reply by nguyenkims** [reply]
Whether an email is encrypted is actually up to the sender. For SimpleLogin, we publish our PGP key and some people use this key to encrypt their emails sent to our address. Please note this is different to the PGP option on SimpleLogin which uses your **mailbox** PGP key to encrypt the emails.

**Reply by leshik** [reply]
Yes, I understand how PGP works and that its use is up to the sender. But my use case is different. Suppose you get some informational emails from a financial institution, such as bank statements. Or trading reports with sensitive data from an exchange. Most of them don't have the option to add PGP encryption to emails, but that's fine - at least SMTP encryption in transit is used. But I would like to prevent Google to spy on this information while it's in my inbox, i.e. some sort of encryption at rest.

**Reply by leshik** [reply]
For Gmail, I solved it now by adding an additional, plus sign mailbox, and by setting the PGP key for it. But it might be useful for other services too. The main obstacle might be is how to deal if the key is set for both an alias and the mailbox. The solution is probably to use an alias key if it's set and to ignore the mailbox one.

**Reply by nguyenkims** [reply]
> But I would like to prevent Google to spy on this information while it's in my inbox, i.e. some sort of encryption at rest.

You can in this case add a PGP key for your SimpleLogin mailbox: emails will be encrypted with this key before being forwarded to your mailbox. This way the email is encrypted in your Gmail.

> The solution is probably to use an alias key if it's set and to ignore the mailbox one.

This is indeed a good idea, can you please create a feature request for it on https://github.com/simple-login/app/discussions so other people can upvote :) ?
